api/src/main/java/uk/gov/dhsc/htbhf/claimant/repository/ClaimantRepository.java

```diff
@@ -5,17 +5,45 @@
 import org.springframework.data.repository.query.Param;
 import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
 
+import java.util.Optional;
 import java.util.UUID;
 
 /**
  * JPA repository for the Claimant table.
  */
 public interface ClaimantRepository extends CrudRepository<Claimant, UUID> {
 
-    @Query("SELECT COUNT(claimant) FROM Claimant claimant WHERE claimant.nino = :nino AND claimant.eligibilityStatus = 'ELIGIBLE'")
-    Long getNumberOfMatchingEligibleClaimants(@Param("nino") String nino);
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.nino = :nino "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long countEligibleClaimantsWithNino(@Param("nino") String nino);
 
-    default Boolean eligibleClaimExists(String nino) {
-        return getNumberOfMatchingEligibleClaimants(nino) != 0;
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.dwpHouseholdIdentifier = :dwpHouseholdIdentifier "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long countEligibleClaimantsWithDwpHouseholdIdentifier(@Param("dwpHouseholdIdentifier") String dwpHouseholdIdentifier);
+
+    @Query("SELECT COUNT(claimant) "
+            + "FROM Claimant claimant "
+            + "WHERE claimant.hmrcHouseholdIdentifier = :hmrcHouseholdIdentifier "
+            + "AND claimant.eligibilityStatus = 'ELIGIBLE'")
+    Long countEligibleClaimantsWithHmrcHouseholdIdentifier(@Param("hmrcHouseholdIdentifier") String hmrcHouseholdIdentifier);
+
+    default boolean eligibleClaimExistsForNino(String nino) {
+        return countEligibleClaimantsWithNino(nino) != 0;
+    }
+
+    default boolean eligibleClaimExistsForHousehold(Optional<String> dwpHouseholdIdentifier, Optional<String> hmrcHouseholdIdentifier) {
+        if (dwpHouseholdIdentifier.isPresent()) {
+            return countEligibleClaimantsWithDwpHouseholdIdentifier(dwpHouseholdIdentifier.get()) != 0;
+        }
+
+        if (hmrcHouseholdIdentifier.isPresent()) {
+            return countEligibleClaimantsWithHmrcHouseholdIdentifier(hmrcHouseholdIdentifier.get()) != 0;
+        }
+
```

Comment:
I think we need to consider the scenario where a claimant moves from CTC to UC within a single cycle. At that point I think it's possible that both HMRC and DWP will return ELIGIBLE with a household identifier. The logic that's in this method might sit better in the claim service - the repository shouldn't really have decision logic.

Reply:
Logic now in service